src/main/java/org/alfresco/bm/rest/SearchContent.java

```diff
@@ -0,0 +1,283 @@
+	private String searchQuery;
+	private String searchType;
+
+	// Search Services properties
+	private String searchLanguage;
+	private int maxItems;
+	private int skipCount;
+
+	// Search Services Range properties
+	private String rangeField;
+	private String rangeStart;
+	private String rangeStop;
+	private String rangeGap;
+	private boolean rangeHardend;
+
+	private String facetQueries;
+
+	// Insight Engine Search properties
+	private String format;
+	private boolean includeMetadata;
+	private String timezone;
+
+	public SearchContent() {
+		this.eventSearchContent = EVENT_SEARCH_CONTENT;
+	}
+
+	public void setEventSearchContent(String eventSearchContent) {
+		this.eventSearchContent = eventSearchContent;
+	}
+
+	public void setUserDataService(UserDataService userDataService) {
+		this.userDataService = userDataService;
+	}
+
+	public void setSearchQuery(String searchQuery) {
+		this.searchQuery = searchQuery;
+	}
+
+	public void setSearchType(String searchType) {
+		this.searchType = searchType;
+	}
+
+	public void setSearchLanguage(String searchLanguage) {
+		this.searchLanguage = searchLanguage;
+	}
+
+	public void setMaxItems(int maxItems) {
+		this.maxItems = maxItems;
+	}
+
+	public void setSkipCount(int skipCount) {
+		this.skipCount = skipCount;
+	}
+
+	public void setRangeField(String rangeField) {
+		this.rangeField = rangeField;
+	}
+
+	public void setRangeStart(String rangeStart) {
+		this.rangeStart = rangeStart;
+	}
+
+	public void setRangeStop(String rangeStop) {
+		this.rangeStop = rangeStop;
+	}
+
+	public void setRangeGap(String rangeGap) {
+		this.rangeGap = rangeGap;
+	}
+
+	public void setRangeHardend(boolean rangeHardend) {
+		this.rangeHardend = rangeHardend;
+	}
+
+	public void setFacetQueries(String facetQueries) {
+		this.facetQueries = facetQueries;
+	}
+
+	public void setFormat(String format) {
+		this.format = format;
+	}
+
+	public void setIncludeMetadata(boolean includeMetadata) {
+		this.includeMetadata = includeMetadata;
+	}
+
+	public void setTimezone(String timezone) {
+		this.timezone = timezone;
+	}
+
+	@Override
+	protected EventResult processEvent(Event event) throws Exception {
+
+		super.suspendTimer();
+
+		DBObject dataObj = (DBObject) event.getData();
+		if (dataObj == null) {
+			throw new IllegalStateException("This processor requires data with field " + SiteContextConstants.SITE_ID);
+		}
+
+		String site = (String) dataObj.get(SiteContextConstants.SITE_ID);
+		String member = (String) dataObj.get(SiteContextConstants.SITE_MEMBER);
+
+		// Get the user for the selected site member
+		UserData userdata = userDataService.findUserByUsername(member);
+		if (userdata == null) {
+			return new EventResult("Failure: The picked user doesn't exist anymore: " + member, false);
+		}
+
+		super.resumeTimer();
+		int count = 0;
+
+		if (searchType != null && searchType.equalsIgnoreCase(SEARCH_SERVICES_SEARCH_TYPE)) {
+			count = performSearchSearvicesSearch(userdata);
+		} else {
```

Comment:
Possibly we could have three cases here and throw an exception if neither SS or IE is matched. This might make it slightly easier to debug issues where SearchServices is mistyped.

Reply:
Fixed it!